### ISSUE TITLE
Face_recognition

### ISSUE THREAD

**Issue by cmlyldz** [davisking/dlib#882]
I am dealing with dnn_face_recognition example it works but I have question about face chip details functions. In this example you use deep learning for face recognition but the algorithm can recognize a face with using 1 training image(face). I want to use 2 different image(posing, or another things) for same person in training. Is that possible ?

**Comment by davisking** [comment]
If you have multiple faces use a k-nearest-neighbor or linear SVM
classifier.

**Comment by cmlyldz** [comment]
Thanks for the answer.

**Comment by Kagami** [comment]
>If you have multiple faces use a k-nearest-neighbor

Sorry for my ignorance but I searched for all mentions of "k-nearest" in dlib's source code and only found 3 suitable functions:

```
find_k_nearest_neighbors
find_k_nearest_neighbors_lsh
find_approximate_k_nearest_neighbors
```

What it does is creates a graph based on distance of provided samples. But if I already have labeled samples and need to find a label for the new sample, how would that help?

Isn't it simpler to just find minimal distance in for loop against all known samples? Sorry if I don't understand something.

For example scikit's KNeighborsClassifier has special [predict](http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html#sklearn.neighbors.KNeighborsClassifier.predict) method.

**Comment by davisking** [comment]
Yes, use a simple for loop.  You don't need to do anything more complex
unless the for loop ends up being too slow.

**Comment by Kagami** [comment]
Thanks, I have one more question. Could `svm_multiclass_linear_trainer` provide better classification results than search for minimal distance? I mean I can always try it, just interested in your opinion because I'm lacking experience in ML.

**Comment by davisking** [comment]
It might be better.  You have to experiment and see what works best
though.  But I wouldn't be surprised at all if it gave better results.
